src/TypeSpec.Extension/Emitter.Csharp/samples/CadlFirstTest/Generated/tspCodeModel.json

```diff
@@ -29,6 +29,7 @@
     }
    ],
    "IsExtensible": false,
+   "IsConfident": true,
```

Comment:
I would have expected the emitter to emit an enum value for what the reason is?  Otherwise don't we have to detect it twice once in emitter and once in generator?

Or we don't have this here at all and do all the detection in the generator might be better?

Reply:
The description should explained this.
How this is implemented:
1. Every type in the emitter will have a `boolean IsConfident`.
2. The `InputOperation` has a property `ConvenienceMethodOmitReason` (enum)
3. The emitter will propagate the IsConfident value on the tree.
4. The generator does nothing, it just reads the reason from the `tspCodeModel.json`. No calculations are done in the generator. Everything is calculated in the emitter.